### ISSUE TITLE
Compression middleware is lost via ApolloEngine.listen

### ISSUE THREAD

**Issue by intellix** [apollographql/apollo-engine-js#155]
It seems like the compression() middleware doesn't apply when passing through the new ApolloEngine. Potentially other middlewares but am not quite sure

```ts
const app = express();

app.use(compression());

engine.listen({
  port: API_PORT,
  graphqlPaths: ['/graphql'],
  expressApp: app,
  launcherOptions: {
    startupTimeout: 3000,
  },
});
```

Either of these correctly respond with: `Content-Encoding: gzip`
```
app.listen(API_PORT);
server.listen(API_PORT);
```

Mentioned this in Slack but guess it might get lost there. Maybe engine itself doesn't relay those headers?

**Comment by sbrichardson** [comment]
I think if Apollo engine is going to be the primary web server we should have access to the full configuration to modify it how needed. Or add back in the other express methods vs only taking .listen()

**Comment by piehei** [comment]
I agree with you ^.

One potential workaround — though of course not suitable for every app — would be to have the separation at a higher level (or maybe lower would be a better word?): 

engine.listen() at port 1
app.listen() at port 2
nginx/apache/network facing server routing based on path

**Comment by sbrichardson** [comment]
@glasser Question - I didn't explore the sidecar/proxy method pre 1.0, is that still an option? Or for engine v1+ is the new strategy of using engine.listen() the preferred (or only) way to use apollo engine?

Also, is the engine proxy source available or viewable? I was trying to investigate modifying the web server options there but only could find a reference. Would it be just a matter of passing normal configuration options to the Go Proxy/Webserver etc. I'm fine in passing options directly, instead of apollo engine needing an intermediary api for advanced cases. 

Just brainstorming options on my end. Thanks again for the work on engine.

Just an FYI - Below is how I was using engine pre-1.0. We are doing a lot of dynamic schema building, multi-tenant custom schemas etc. Everything is working solid, I updated for 1.0 but reverted back temporarily.

```js
export default async () => {

  //////////////////////////
  // Configure Middleware //
  //////////////////////////

  /** Startup database to cache client */
  await startDatabase()

  /** Initiate express, apollo engine and middleware */
  await engine.start()

  /** Instantiate express app */
  const app = express()

  /** Add CORS security settings */
  const CORSConfig = { origin: CORS_WHITELIST }
  app.use(cors(CORSConfig))

  /** Process Apollo Engine */
  app.use(engine.expressMiddleware())

  /** Process compression settings */
  const compressionConfig = {}
  app.use(compression(compressionConfig))

  /** Start System Change Streams */
  startSystemStreams()

  ///////////////////
  // Define Routes //
  ///////////////////

  /** GraphQL API Route */
  app.use(GQL_SERVER_ENDPOINT, json(), auth, getSchema, routes_graphql)

  /** rCall route */
  app.use('/rcall', json(), auth, routes_rcall)

  /** Health check route for monitoring */
  app.get('/health', routes_health)

  //////////////////
  // Start Server //
  //////////////////

  app.listen(PORT, () => console.info(`:${PORT}  started.`))
}
```

**Comment by glasser** [comment]
@intellix Thanks for the report. We should definitely update engineproxy to support content-encoding:gzip (on by default, disable-able if necessary.) It's a priority for me this week.

@sbrichardson Agreed that we need to expose as many knobs as possible into the web server given that it proxies all your web traffic now. The proxy is built on the Go standard HTTP server, client, and proxy libraries, and we've been pretty happy using it in production for years on various projects, but we should provide full accessi to configurations.

We had a lot of problems from the original double-proxying deployment defaults and we hope that single-proxying solves many of our users' problems.  We also suggest that when possible, keeping your GraphQL server separate from other servers can have many benefits. If you do need to co-host your GraphQL endpoints on the same server as other behavior and passing that through Engine's proxy is untenable (even once we add additional configuration knobs) it might be best to, as you and @piehei suggest, configure nginx or another router to split traffic between Engine (configured manually with ApolloEngineLauncher or Docker) and your app directly.  However, we hope that the ApolloEngine API works well for most Node users.

**Comment by glasser** [comment]
You could manually run a compressing proxy in front of your server.

A fix is currently in progress; hopefully out this week but there's another priority in front of it now.

Note that the lack of compression only applies to GraphQL responses, not other proxied HTTP traffic. (If anyone thinks that is not accurate, I'd be curious.)

**Comment by glasser** [comment]
Fixed in 1.0.5. https://www.apollographql.com/docs/engine/proxy-release-notes.html
